### ISSUE TITLE
Allow vouchers to have stock

### ISSUE THREAD

**Issue by pixelmachine** [verbb/gift-voucher#105]
### What are you trying to do?

I've been asked a couple of times recently to setup limited runs of vouchers. 

It's not too hard to add a basic implementation via a custom field and an event but it would be nicer if vouchers had the same stock options as full products, i.e. stock or unlimited stock. This would bring vouchers more inline with standard Commerce products, which is nice for clients, and allows simpler templating an so on.

Not sure if that's something that can be inherited from the built in elements.

### What's your proposed solution?

Stock level field and unlimited option. Unlimited by default I guess.

**Comment by engram-design** [comment]
Yep, all the pieces are there thanks to Commerce purchasable, I just didn't imagine a use-case for limited digital vouchers, but sounds like that's valid.

**Comment by pixelmachine** [comment]
I think Black Friday has my clients getting creative. Also they're hospitality based so vouchers are often for types of services which may need to be limited.
